### ISSUE TITLE
`readonly ref` is changed to `ref readonly` causing error CS9190

### ISSUE THREAD

**Issue by aurnoi1** [belav/csharpier#1123]
Hi, 

When running chsharpier, `readonly ref` is changed to `ref readonly` causing error [CS9190](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/ref-modifiers-errors?f1url=%3FappId%3Droslyn%26k%3Dk(CS9190)#incorrect-syntax).

Version dotnet: net8.0
Version csharpier: 0.26.7

**Input:**

```csharp
    public static Result<int> DoSomething(ref readonly Result<int> result)
    {
        return result;
    }
```

**Output:**
```csharp
    public static Result<int> DoSomething(readonly ref Result<int> result)
    {
        return result;
    }
```

**Expected behavior:**
```csharp
    public static Result<int> DoSomething(ref readonly Result<int> result)
    {
        return result;
    }
```

Thanks!

**Comment by belav** [comment]
I modified the sorting modifiers logic to only run on specific nodes to hopefully prevent something like this in the future. Thanks for reporting the problem, it'll be fixed in the coming release!

**Comment by aurnoi1** [comment]
I confirmed that the but is fixed on my side in the last version 0.27.1. Thanks!
